src/AmznSPAConfig.php

```diff
@@ -48,6 +51,9 @@ public function __construct(
         ?CarbonImmutable $grantless_access_token_expires_at = null,
         ?string $restricted_data_token = null,
         ?CarbonImmutable $restricted_data_token_expires_at = null,
+        ?bool $use_proxy = false,
```

Comment:
Can we use a Proxy value object instead? Then we can have a single additional parameter here which is a nullable Proxy object that contains the url and auth token.

Reply:
ready, see the next commit